docs/technical_reference/unit_models/stoichiometric_reactor.rst

```diff
@@ -0,0 +1,198 @@
+of the feed stream, using ions present in the feed. 
+
+Example dictionary for dissolving Soda ash and Lime into their ions  
+
+   .. code-block:: 
+
+      reagents = {
+            "Na2CO3": {
+                  "mw": 105.99 * pyunits.g / pyunits.mol,
+                  "dissolution_stoichiometric": {"Na_+": 2, "HCO3_-": 1},
+            },
+            "CaO": {
+                  "mw": 56.0774 * pyunits.g / pyunits.mol,
+                  "dissolution_stoichiometric": {"Ca_2+": 1, "H2O": 1},
+            },
+         }
+
+Example dictionary for defined precipitant that would form during reaction (Calcite - CaCO3, and Brucite - MgOH)
+
+   .. code-block:: 
+
+      precipitants = {
+         "Calcite": {
+               "mw": 100.09 * pyunits.g / pyunits.mol,
+               "precipitation_stoichiometric": {"Ca_2+": 1, "HCO3_-": 1},
+         },
+         "Brucite": {
+               "mw": 58.3197 * pyunits.g / pyunits.mol,
+               "precipitation_stoichiometric": {"Mg_2+": 1, "H2O": 1},
+         },
+      }
+
+The user can then provide the model with specified dictionary to produce a stoichiometric reactor that only 
+perform dissolution, precipitation, or both. 
+
+   .. code-block:: 
+
+      # unit for only adding a reagent 
+      m.fs.chemical_addition = StoichiometricReactor(
+            property_package=m.fs.properties,
+            reagent=reagents,
+         )
+      # The user must the specify how much reagent to add
+      m.fs.chemical_addition.reagent_dose["Na2CO3"].fix(1e-3)
+      m.fs.chemical_addition.reagent_dose["CaO"].fix(1e-3)
+      
+      # unit for only precipitating specified species out of the feed
+      m.fs.precipitation_reactor = StoichiometricReactor(
+            property_package=m.fs.properties,
+            precipitants=precipitants,
+         )
+      # The user must then specify how much precipitant to form 
+      m.fs.precipitation_reactor.flow_mass_precipitate["Calcite"].fix(1e-3)
+      m.fs.precipitation_reactor.flow_mass_precipitate["Brucite"].fix(1e-4)
+      # The user must also specify solids fraction in waste stream
+      m.fs.unit.waste_mass_frac_precipitate.fix(0.2)
+      
+      # unit for addition and precipitation (e.g. traditional Lime/Soda ash softening process)
+      m.fs.lime_soda_softening = StoichiometricReactor(
+            property_package=m.fs.properties,
+            reagent=reagents,
+            precipitants=precipitants,
+         )
+      # The user must the specify how much reagent to add and precipitant to form
+      m.fs.lime_soda_softening.reagent_dose["Na2CO3"].fix(1e-3)
+      m.fs.lime_soda_softening.reagent_dose["CaO"].fix(1e-3)
+      m.fs.lime_soda_softening.flow_mass_precipitate["Calcite"].fix(1e-3)
+      m.fs.lime_soda_softening.flow_mass_precipitate["Brucite"].fix(1e-4)
+      # The user must also specify solids fraction in waste stream
+      m.fs.lime_soda_softening.waste_mass_frac_precipitate.fix(0.2)
```

Comment:
This bit seems more like a "[how-to](https://diataxis.fr/how-to-guides/)" rather than [technical reference](https://diataxis.fr/reference/). We've been trying to follow the Diataxis framework for documenation,

Reply:
I figured this was the clearest way to provide a reference to the model. And other docs provide code examples as well...  This can be sorted out in another PR if we want to have a separate how-to...